docs/how-tos/how-to-use-searchkick-for-search-autocomplete.mdx

```diff
@@ -0,0 +1,285 @@
+
+- `searchkick word_start: [:name]` enables Searchkick for this
+  model. `word_start: [:name]` will index the name of the product
+  in a special way that allows a partial match with the initial
+  part of the string. By default, it matches the entire word.
+  We will see later why we need this.
+- `search_data` tells Searchkick which fields we want to index.
+  For this example, `name` is enough but you could consider adding
+  more fields depending on your needs.
+- `should_index?` determines which records will be indexed. In this
+  case, to keep it simple we only index products that are not
+  soft-deleted.
+
+Now that we've got the basis, it's time for our first real indexing.
+Opening a Rails console, we can run:
+
+```ruby title="bin/rails c"
+Spree::Product.reindex
+```
+
+:::info
+
+Be sure you have Elasticsearch up and running. Please, follow the
+official [installation instructions][elasticsearch-install-guide],
+or if you are a MacOS and Homebrew user, just run:
+
+```sh
+brew install elastic/tap/elasticsearch-full
+brew services start elasticsearch-full
+```
+
+:::
+
+At this poing we can verify that a basic search is working.
+Assuming you have some products which include the `ruby`
+word in one of the indexed fields, you can run:
+
+```ruby title="bin/rails c"
+Spree::Product.search("ruby").map(&:name)
+# => ["Ruby Hoodie", "Ruby Polo", "Ruby Mug", "Ruby Tote", "Ruby Hoodie Zip"]
+```
+
+Yey, it works! 🎉
+
+Time to index our categories as well:
+
+```ruby title="app/overrides/my_store/spree/taxon/add_searchkick.rb"
+module AmazingStore
+  module Spree
+    module Taxon
+      module AddSearchkick
+        def self.prepended(base)
+          unless base.respond_to?(:searchkick_index)
+            base.searchkick word_start: [:name]
+          end
+        end
+
+        def search_data
+          { name: name }
+        end
+
+        ::Spree::Taxon.prepend self
+      end
+    end
+  end
+end
+```
+
+We can now run a reindex for `Spree::Taxon` and verify it's
+working:
+
+```ruby title="bin/rails c"
+Spree::Taxon.reindex
+Spree::Taxon.search("ruby").map(&:name)
+# => ["Ruby"]
+```
+
+## Enhancing search autocomplete
+
+[Solidus Starter Frontend][ssf] comes with a simple search autocomplete out of
+the box. In this section of the guide we will change its behavior to use the
+newly added advanced search functionalities.
+
+:::info
+
+The search autocomplete feature has been added to the Starter Frontend
+since Solidus 3.4.0. If you don't have it, please refer to
+[Autocomplete main search with products and taxons][autocomplete-pr] to
+backport that feature in your storefront.
+
+:::
+
+To let the autocomplete controller use Searchkick, we need to change the
+code of the `AutocompleteResultsController` provided like this:
+
+```diff title="app/controllers/autocomplete_results_controller.rb"
+
+  def autocomplete_products
+    if params[:keywords].present?
+-      searcher = build_searcher(params.merge(per_page: 5))
```

Comment:
What do you think about showing how to configure a custom `Spree::Config.searcher_class` instead? That would be an excellent way to showcase Solidus' flexibility, plus we'd avoid touching the controller code.

Reply:
That's a good point, and I tried coming back to this simpler solution; let me explain why: if we change the searcher we need to support with Searchkick all the possible params that are supported by the current ActiveRecord version, including all the logic related to prices validity, stock availability, taxon inclusion. Plus, the search has the support to [search by scope](https://github.com/solidusio/solidus/blob/17540ad0c9d7c8d69d7913a0942edf70e9e8f183/core/lib/spree/core/search/base.rb#L65), which makes the potential fields to be indexed almost infinite, because it depends on the scope you have and want to support.

That said, we could create a hybrid version of the searcher, changing only the part that refers to the [plain search](https://github.com/solidusio/solidus/blob/17540ad0c9d7c8d69d7913a0942edf70e9e8f183/core/lib/spree/core/search/base.rb#L82-L88). But honestly I'm not convinced that this is really helpful because you will still use AR for the rest of filtering logics, and we would relegate Searchkick to do something that you can already do with AR. 

My idea here was to show an easy way to introduce searchkick for specifically for the autocomplete, to reduce the complexity of the tutorial. But it's clear that people may expect the main product searcher to act similarly to the results provided by the autocomplete. Thinking that we could explain that if that's the case they have to build their own searcher class based on Searchkick.

Reply:
I see. I agree that a simple comment would be enough here. Ideally, we'd create a how-to for custom searchers at some point and add a link from here.

Reply:
I added a final paragraph about this in the Next Steps section. Please, let me know if it's more clear now.

Reply:
Thanks, awesome! Thanks, @kennyadsl!